### ISSUE TITLE
Feature request: Whitelist of App ID

### ISSUE THREAD

**Issue by ngoclong19** [Rudokhvist/ASF-STM#17]
For users with a large number of badges, it would take a very long time to scan all their badges and find matching bots.

I would like to suggest adding a whitelist for app ID.
So if a user want to complete a specific `n` game badges, they could only need to search for `n x M` times, instead of the current `N x M`, where `N` is the number of available badges, and `M` is the number of bots.

This pre-search whitelist would come in handy with the current after search filter.

**Comment by Rudokhvist** [comment]
Yes, that's a good idea. Can't promise to make it soon, don't have much free time recently, but definitely will add to wishlist. And of course PRs are welcome.

**Comment by ngoclong19** [comment]
I will craft a PR later.

**Comment by iBreakEverything** [comment]
I wrote something for this, but I didn't add UI interactions for it
add this line near the top after `"use strict";` and put inside appIDs you want to look for
```
    let filters = [730, 1085750, 1849000];
```
before GetCards(0, -1); line (line no ~1274) insert this:
```
                                if (filters.length > 0) {
                                    myBadges = myBadges.filter(x => filters.includes(x.appId));
                                }
```

script will only search for `filters` appIDs you provide

**Comment by iBreakEverything** [comment]
@ngoclong19 Scan filters are now a feature in [✨ASF-STM-Enhancement✨](https://github.com/iBreakEverything/ASF-STM-Enhancement)

**Comment by ngoclong19** [comment]
Thank you, @iBreakEverything! Those DEV improvements look interesting.
